### DISCUSSION TITLE
Move support for third-party plugins into separate plugins

### DISCUSSION THREAD

**Discussion by BenParizek** [engram-design/FeedMe#898] [Ideas]
It's nice that FeedMe supports several third-party plugins:
https://docs.craftcms.com/feed-me/v4/content-mapping/element-types.html
https://docs.craftcms.com/feed-me/v4/content-mapping/field-types.html#third-party

As individual plugin developers, sometimes we choose to have integrated support for other third-party plugins in the ecosystem.

However, now that FeedMe is a first-party plugin all third-party support should be moved to separate plugins that integrate with the documented FeedMe APIs, as not to favor some plugins in the ecosystem over others within a first-party tool.

While I prefer support for all third-party integrations to be handled via separate plugins via a documented API, if FeedMe supports some third-party plugins as a feature there should be a documented process for other third-party plugins to also be added as features.

**Comment by lindseydiloreto** [comment]
In Feed Me 1, third-party support was actually handled _within_ the third-party plugin (via hooks). That approach has always seemed more logical to me, since the mechanics may change between plugin versions.

Additionally, this would allow custom modules to register their own import mechanisms as well.

**Comment by angrybrad** [comment]
@BenParizek this is 100% the plan
